.github/workflows/release.yaml

```diff
@@ -0,0 +1,90 @@
+
+      - name: Debug version
+        run: echo ${{ env.VERSION }}
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login Helm to DockerHub registry
+        run: |
+           helm registry login registry-1.docker.io -u ${{ secrets.DOCKERHUB_USERNAME }} --password-stdin <<< "${{ secrets.DOCKERHUB_TOKEN }}"
+
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Build and publish Docker image (init container)
+        env:
+          VERSION: ${{ env.VERSION }}
+        run: |
+          echo "Building Docker init image with version: ${VERSION}"
+          docker build -t kubeshop/monokle-admission-controller-init:latest -t kubeshop/monokle-admission-controller-init:${VERSION} ./admission-controller/init/
+          docker push kubeshop/monokle-admission-controller-init:${VERSION}
+          docker push kubeshop/monokle-admission-controller-init:latest
+
+      - name: Build and publish Docker image (server container)
+        env:
+          VERSION: ${{ env.VERSION }}
+        run: |
+          echo "Building Docker server image with version: ${VERSION}"
+          docker build -t kubeshop/monokle-admission-controller-server:latest -t kubeshop/monokle-admission-controller-server:${VERSION} ./admission-controller/server/
+          docker push kubeshop/monokle-admission-controller-server:${VERSION}
+          docker push kubeshop/monokle-admission-controller-server:latest
+
+      # See https://docs.docker.com/docker-hub/oci-artifacts/#push-a-helm-chart
+      - name: Build and publish Helm chart
+        env:
+          VERSION: ${{ env.VERSION }}
+        run: |
+          echo "Building Helm chart with version: ${VERSION}"
+          helm package ./helm --version ${VERSION}
+          helm push monokle-admission-controller-${VERSION}.tgz oci://registry-1.docker.io/kubeshop
+
+  create-release:
+    runs-on: ubuntu-latest
+    needs: build-and-push
+
+    steps:
+      - name: Extract version from tag
+        id: extract_version
+        run: echo "VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_ENV
+
+      - name: Debug version
+        run: echo ${{ env.VERSION }}
+
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Build release artifacts
+        env:
+          VERSION: ${{ env.VERSION }}
+        run: |
+          helm package ./helm --version ${VERSION}
+          helm template ./helm --version ${VERSION} > install.yaml
+          mv monokle-admission-controller-${VERSION}.tgz helm.tgz
+
+      - name: Create release and upload artifacts
+        uses: ncipollo/release-action@v1
+        with:
+          name: "Release ${{ env.VERSION }}"
+          allowUpdates: true
+          artifacts: 'install.yaml,helm.tgz'
+          omitBody: true
+          token: ${{ secrets.CI_BOT_TOKEN }}
```

Comment:
Similarly, this could be a separate job for creating the manifest release only on changes to the manifest files (or on demand).

Reply:
Probably the answer will be the same as in the comment above:

> The idea was to have separate version for each release.

And here, creating manifest seems pretty cheap/fast. I guess reusing ones (based on changes) would be very similar when it comes to time/resources consumed? Unless you had something different in mind?

Reply:
Ok, maybe because I went with, kind of, coupling manifest version with container versions. Still, if any container version is updated and released, manifests needs to be released to point to latest images.

The other way around not necessarily, so changes in manifests (Helm chart) only, should't result in new docker images. Probably that's what you had in mind? 🤔